lib/tutorials/en_US/cookies.md

```diff
@@ -1,59 +1,98 @@
+
+We then tell hapi that the value is a base64 encoded JSON string.
+
+By setting `clearInvalid` to `true`, it instructs the client to remove invalid cookies.
+
+Last, we set `strictHeader` to `true` so that there are no violations of [RFC 6265](https://tools.ietf.org/html/rfc6265).
 
-This configuration will make it so the cookie named `data` has a session time-life (will be deleted by the browser when closed), is flagged both secure and HTTP only (see [RFC 6265](http://tools.ietf.org/html/rfc6265), specifically sections [4.1.2.5](http://tools.ietf.org/html/rfc6265#section-4.1.2.5) and [4.1.2.6](http://tools.ietf.org/html/rfc6265#section-4.1.2.6) for more details about these flags), and tells hapi that the value is a base64 encoded JSON string. Full documentation for the `server.state()` options can be found in [the API reference](/api#serverstatename-options).
+### <a name="options.state" /> route.options.state()
 
-In addition to this, you can further configure cookie behaviour at a route-level by specifying two properties at the route's `options.state` object:
+In addition to this, you can further configure cookie behavior at a route-level by specifying two properties at the route's `options.state` object:
 
 ```json5
 {
     options: {
         state: {
-            parse: true,        // parse cookies and store in request.state
-            failAction: 'error' // may also be 'ignore' or 'log'
+            parse: true,
+            failAction: 'error'
         }
     }
 }
 ```
+The `parse` option determines if cookies are parsed and stored in `request.state`.
 
-## Setting a cookie
+The `failAction` options determines how cookie parsing errors will be handled.
 
-Setting a cookie is done via the [response toolkit](/api#response-toolkit) in a request handler, pre-requisite, or request lifecycle extension point and looks like the following:
+## <a name="setting" /> Setting a cookie
+
+Setting a cookie is done via the [response toolkit](/api#response-toolkit) in a request handler, pre-requisite, or request lifecycle extension point.
+
+### <a name="h.state" /> h.state()
+
+We set a cookie by calling [`h.state(name, value, [options]`](https://hapijs.com/api#h.state()). In the following example, we set a cookie in a route handler:
 
 ```javascript
-h.state('data', { firstVisit: false });
-return h.response('Hello');
-```
+server.route({
+    method: 'GET',
+    path: '/',
+    handler: function (request, h) {
 
-In this example, hapi will reply with the string `Hello` as well as set a cookie named `data` to a base64 encoded JSON string representation of `{ firstVisit: false }`.
+        h.state('data', { firstVisit: false });
+        return h.response('Hello');
+    }
+});
+```
+In this example, we use the cookie we configured in the tutorial above. Here, hapi will reply with the string `Hello` as well as set a cookie named `data` to a base64 encoded JSON string representation of `{ firstVisit: false }`.
```

Comment:
My gut feeling is that “in the tutorial above” is wrong. I’d say it’s the section above, is it?